vm/systemSmartContracts/delegation.go

```diff
@@ -1108,10 +1117,43 @@ func (d *delegation) updateRewards(args *vmcommon.ContractCallInput) vmcommon.Re
 		return vmcommon.UserError
 	}
 
-	return vmcommon.UserError
+	return vmcommon.Ok
+}
+
+func (d *delegation) getRewardData(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if len(args.Arguments) != 1 {
+		d.eei.AddReturnMessage("must call with 1 arguments")
+		return vmcommon.UserError
+	}
+	if args.CallValue.Cmp(zero) != 0 {
+		d.eei.AddReturnMessage("cannot call with negative value")
+		return vmcommon.UserError
+	}
+	err := d.eei.UseGas(d.gasCost.MetaChainSystemSCsCost.DelegationOps)
```

Comment:
why do we need to call getRewardData from the wallet? Why this is not a view function? Maybe we should consider making all view functions callable from the owner or the auction/staking SC as someone will want to send a transaction and notarize the status of the contract at a certain moment in time (similar to a printed receipt done by an ATM)

Reply:
some view functions might be called by users as well. thus they need some cost. others are too computationally expensive and are made with view function protection by default.